pennylane/qnode.py

```diff
@@ -505,26 +509,22 @@ def _op_successors(self, o_idx, only='G'):
                 - ``'G'``: only return non-observables (default)
                 - ``'E'``: only return observables
                 - ``None``: return all successors
+            get_nodes (bool): if True, returns nodes instead of operations
 
         Returns:
             list[Operation]: successors in a topological order
         """
-        succ = self.ops[o_idx+1:]
-        # TODO at some point we may wish to upgrade to a DAG description of the circuit instead
-        # of a simple queue, in which case,
-        #
-        #   succ = nx.dag.topological_sort(self.DAG.subgraph(nx.dag.descendants(self.DAG, op)).copy())
-        #
-        # or maybe just
-        #
-        #   succ = nx.dfs_preorder_nodes(self.DAG, op)
-        #
-        # if it is in a topological order? the docs aren't clear.
+        succ = self.circuit.get_nodes(self.circuit.descendants((o_idx,)))
+
         if only == 'E':
```

Comment:
We might want to update the argument to be 'O' (observable) instead of 'E' (expectation, now deprecated usage to refer to operations)

Reply:
See #282